src/FastRandom.cpp

```diff
@@ -6,12 +6,28 @@
 #include "Globals.h"
 #include "FastRandom.h"
 
+#include <random>
+
 #ifdef _WIN32
-	#define thread_local __declspec(thread)
+	#define thread_local static __declspec(thread)
+#elseif __OSX__
+	#define thread_local static __thread
 #endif
 
-thread_local unsigned int m_Counter = 0;
-
+static unsigned int GetRandomSeed()
+{
+	thread_local bool SeedCounterInitialized = 0;
```

Comment:
Yes, but I wanted a initialization value I was sure all three different thread_local systems would accept, and zero initialization is the safest there is.